device/chip/tlb_manager.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2024 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include "umd/device/tt_xy_pair.h"
+#include "umd/device/types/tlb.h"
+
+namespace tt {
+class Writer;
+}
+
+namespace tt::umd {
+
+class TTDevice;
+
+class TLBManager {
+public:
+    TLBManager(TTDevice* tt_device);
+
+    void configure_tlb(tt_xy_pair core, int32_t tlb_index, uint64_t address, uint64_t ordering);
```

Comment:
How does the caller know what a valid tlb_index is?

Reply:
A very good question, i have to think about it

Reply:
It doesn't, that is one of the flaws of how UMD manages TLBs today. That is why we would like to transition to different API, something along the lines of

```
TLBWindow* get_tlb_wc();
TLBWindow* get_tlb_uc();
TLBWindow* get_any_tlb();
```

Note: TLBWindow is pointing to the struct that Joel suggested in some of the issues earlier.

then the client doesn't need to worry about index. We could however add a check baed on architecture implementation

Reply:
Wrote a test and added a couple of utility functions to architecture_implementation.
Ultimatelly, the workload user (using chip and cluster) should not see tlb_index in their APIs, will work on removing that. tlb_index should be surfaced only in the lower levels.

Please take a look at new file test_tlb_manager.cpp